### PR TITLE
docs(idempotency): bring your own persistent store

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

This PR introduces a new section to the docs that discusses how to build your own Persistence Layer for the Idempotency utility.

The section follows closely the structure of the same section found in [the Python docs](https://docs.powertools.aws.dev/lambda/python/latest/utilities/idempotency/#bring-your-own-persistent-store) but instead it shows how to implement the alternative persistence layer with a generic key-value store.

The PR also:
- extracts all the JSON events and infra snippets (i.e. SAM, Terraform, etc.) to their own dedicated files so that they can be linted/formatted
- adds a missing CDK sample to show how to create the table required for Idempotency

### Related issues, RFCs

**Issue number:** #1484

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [ ] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: docs/snippets/idempotency/templates/tableCdk.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const table = new Table(this, 'idempotencyTable', {
+      partitionKey: {
+        name: 'id',
+        type: AttributeType.STRING,
+      },
+      timeToLiveAttribute: 'expiration',
+      billingMode: BillingMode.PAY_PER_REQUEST,
+    });
+
+    const fnHandler = new NodejsFunction(this, 'helloWorldFunction', {
```
Comment (am29d) [inline docs/snippets/idempotency/templates/tableCdk.ts:20]: can we add `entry` and `handler` as a place holder? I understand that `NodejsFunction` can be smart for specific file and resource name convention. But it'd be great to be more explicit for the folks who are new.

**Review by am29d — CHANGES_REQUESTED** [review]
Looks great, just a small change for the CDK example.

File: docs/snippets/idempotency/templates/tableCdk.ts (hunk 1 above)
Comment (dreamorosi) [reply docs/snippets/idempotency/templates/tableCdk.ts:20]: Good point, I'll add it later today

**Comment by dreamorosi** [comment]
Comments addressed, merging so I can unblock another PR and avoid merge conflicts on `package-lock.json`.

**Pull request merged**